client/src/index.tsx

```diff
@@ -10,6 +10,87 @@ import { GleanProvider } from "./telemetry/glean-context";
 
 // import * as serviceWorker from './serviceWorker';
 
+let attemptedFallback = false;
+window.addEventListener("error", async (error) => {
+  const root = document.querySelector("#root");
+  if (
+    !attemptedFallback &&
+    root &&
+    // only fall back if react gave up and unmounted, leaving an empty page
+    root.innerHTML.trim() === ""
+  ) {
+    attemptedFallback = true;
+    try {
+      const res = await fetch(window.location.pathname, {
+        cache: "force-cache",
+      });
+      const page = await res.text();
+      const parser = new DOMParser();
+      const fallbackRoot = parser
+        .parseFromString(page, "text/html")
+        .querySelector("#root");
+      if (fallbackRoot) {
+        root.innerHTML = fallbackRoot.innerHTML;
+        console.warn(
+          "Fell back to static HTML page due to unhandled error,",
+          "please report an issue on GitHub:",
+          generateGithubIssueLink(error.error)
+        );
+      } else {
+        throw new Error("no root");
+      }
+    } catch (fallbackError) {
+      console.warn(
+        "Failed falling back to static HTML page:",
+        fallbackError,
+        "please report an issue on GitHub:",
+        generateGithubIssueLink(error.error, fallbackError)
+      );
+    }
+  }
+  // Unlike other events, the error event is canceled by returning true from the handler instead of returning false.
```

Comment:
This confused me at first, because I didn't see why we would _not_ want to cancel the error. 

The docs enlighted me:

> When canceled, the error won't appear in the console, but the current script will still stop executing.

So we want to _not_ cancel the error, **so that** the (original) error message is shown in the console.

But couldn't we just omit the return altogether (like the MDN examples do)?